util/perl/TLSProxy/Proxy.pm

```diff
@@ -414,9 +415,14 @@ sub clientstart
         $pid = $self->{serverpid};
         die "serverpid is zero\n" if $pid == 0;
         print "Waiting for server process to close: $pid...\n";
-        # recall that we wait on process that buffers server's output
-        waitpid($pid, 0);
-        die "exit code $? from server process\n" if $? != 0;
+        # note that we wait on process that buffers server's output
+        $pid = waitpid($pid, 0);
+        if ($pid > 0) {
+            die "exit code $? from server process\n" if $? != 0;
```

Comment:
is this the exit code from the s_server or from the "perl -ne print" ?

Reply:
Commentary says it, "perl -ne print". Yes, one can argue that it's kind meaningless, but this message always was about abnormal cases, and non-zero from perl -ne print is as abnormal as anything else. Primary purpose of this waitpid is synchronization, and synchronization on output. Goal is to prevent interleaving with TAP output, which would confuse TAP parser. But one can say that waiting on s_server pid might be appropriate for the purpose of printing its return value... Is this what you trying to say?

Reply:
This was previously the exit code from openssl s_server ?

Reply:
Yes. Well, message is admittedly misleading... I can fix that. But before I do, answer the question. Do you suggest to collect even s_server's exit value? Note that it's used in with die, i.e. we are looking at condition that was never actually observed... Well, it's kind of poor excuse, I know...

Reply:
I think we are __only__ interested in the exit code from the s_server process.
the print process only needs to be collected, to prevent zombie processes.

Reply:
I mean if the server crashes in the atexit function that is something we should consider.

Reply:
Addressed. Thanks for feedback.